src/community/schemaless-features/schemaless-core/src/main/java/org/geoserver/schemalessfeatures/data/SchemalessFeatureSource.java

```diff
@@ -28,10 +30,12 @@ public FeatureType getSchema() {
         FeatureType featureType = null;
         if (featureType == null) {
             GeometryDescriptor descriptor = getGeometryDescriptor();
+            List<PropertyDescriptor> descriptorList = new ArrayList<>();
+            descriptorList.add(descriptor);
```

Comment:
If this List can be inmutable, you can create the new inmutable List directly using ```Arrays.asList(descriptor)```, or if only one element is needed: ```Collections.singletonList(descriptor)```.

Reply:
the list is mutable since the featureType is a dynamic one and can add descriptors